server/src/main/java/org/candlepin/resource/ProductResource.java

```diff
@@ -289,26 +296,22 @@ public void deleteProduct(
             return null;
         }
 
-        // TODO:
-        // Replace this with the commented out block below once the job scheduling is no longer performed
-        // via PinsetterAsyncFilter
         ResultIterator<Owner> iterator = this.ownerCurator.getOwnersWithProducts(productUuids).iterate();
-        List<JobDetail> details = new LinkedList<>();
+        List<JobConfig> jobConfigs = new LinkedList<>();
         while (iterator.hasNext()) {
-            details.add(RefreshPoolsJob.forOwner(iterator.next(), lazyRegen));
+            JobConfig config = RefreshPoolsJob.createJobConfig()
+                .setOwner(iterator.next())
+                .setLazyRegeneration(lazyRegen);
+            jobConfigs.add(config);
         }
         iterator.close();
 
-        return details.toArray(new JobDetail[0]);
-
-        // final Boolean lazy = lazyRegen; // Necessary to deal with Java's limitations with closures
-        // return this.ownerCurator.lookupOwnersWithProduct(productUuids).transform(
-        //     new ElementTransform<Owner, JobDetail>() {
-        //         @Override
-        //         public JobDetail transform(Owner owner) {
-        //             return RefreshPoolsJob.forOwner(owner, lazy);
-        //         }
-        //     }
-        // );
+        List<AsyncJobStatus> statuses = new LinkedList<>();
+        for (JobConfig config : jobConfigs) {
+            statuses.add(this.jobManager.queueJob(config));
+        }
+
+        return statuses.stream().map(
```

Comment:
random musings:
I wonder how we should handle the situation where one or more jobs are queued successfully and one or more jobs fail queuing. I don't think the pinsetter code handled this case, either, and it hasn't been a problem [yet/that we know of], but this seems like an awkward problem waiting to happen.

Reply:
I also thought about it but couldn't think of what would be a good way. The old job framework would call `scheduleJob()` in a loop for each `JobDetail` in the array, which simply throws a 503 error (so it didn't handle this case at all):
```
    private JobStatus scheduleJob(JobDetail detail) {
        try {
            return this.pinsetterKernel.scheduleSingleJob(detail);
        }
        catch (PinsetterException e) {
            throw new ServiceUnavailableException("Error scheduling refresh job.", e);
        }
    }
```

Reply:
@Ceiu I've worked on handling this case, but I want to run it by you before pushing: http://pastebin.test.redhat.com/769504

Another approach to this would be to add a `Collection<AsyncJobStatus> queueJobs(Collection<JobConfig>)`  method to `JobManager`, which handles the exceptions of each queueing itself and returns a collection of statuses so that the caller doesn't have to do that work each time. (which sounds better than my initial approach).

Reply:
It's not something critical to handle here, as the queueJob method handles the majority of actual issues we're concerned with. As far as valiadation failures go, I'm okay with the pastebin example except for the FAILED_VALIDATION state, as that's a reason for a failure state, not a state itself.

Anyway, splitting of hairs. If you want to add the extra exception handling and output of a generated AsyncJobStatus in a dead state, that's fine. But, again, no new state please. ;)

Reply:
I'll add the example from the pastebin (without new state), since we probably don't have many examples like this one to warrant a new method in JobManager.